cassis/typesystem.py

```diff
@@ -604,10 +612,23 @@ def is_primitive_array(self, type_name) -> bool:
         """
         if type_name == TOP_TYPE_NAME:
             return False
-        elif type_name in _PRIMITIVE_ARRAY_TYPES:
-            return True
-        else:
-            return self.is_primitive_array(self.get_type(type_name).supertypeName)
+
+        # Arrays are inheritance-final, so we do not need to check the inheritance hierarchy
+        return type_name in _PRIMITIVE_ARRAY_TYPES
+
+    def is_array(self, type_name) -> bool:
+        """Checks if the type identified by `type_name` is an array.
+
+        Args:
+            type_name: The name of the type to query for.
+        Returns:
+            Returns True if the type identified by `type_name` is an array type, else False
```

Comment:
`True`

Reply:
What do you mean?

Reply:
I use markdown 'True' `True` instead of True for constants and such